### PR TITLE
Fix highlight background color in codemirror input in dark theme

### PR DESCRIPTION
**Pull request by eseliger**
The search query input highlight background color is hard to see in dark mode. This is because `.theme-dark` CodeMirror theme option isn't being applied, because it namespaces the css classnames. This fixes the color by using a CSS variable that responds to the theme change, and uses a more visible and differentiated color (which also matches the code editor)

| Before | After |
|-|-|
| Screenshot 2023-05-16 at 10 17 20 am | Screenshot 2023-05-16 at 10 19 06 am |

## Test plan

- Open the search query input in multiple places (e.g. home page, on a repo)
- Highlight text
- Focus and unfocus window
- Change theme from light and dark
- Inspect highlight

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.00 kb) | 0.00% (+0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/es+tl/backdrop-color-fix/compare-report.html;tab=live_object) for a full comparison between the commits 3fd4a93 and 812b229 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/branded/src/search-ui/input/experimental/CodeMirrorQueryInputWrapper.tsx [hunk 1]
```diff
@@ -239,7 +239,7 @@ const staticExtensions: Extension = [
         '.cm-line': {
             padding: 0,
         },
-        '.theme-dark .cm-selectionLayer .cm-selectionBackground': {
+        '.theme-dark & .cm-selectionLayer .cm-selectionBackground': {
```
Comment (toolmantim) [inline client/branded/src/search-ui/input/experimental/CodeMirrorQueryInputWrapper.tsx:242]: Oh nice! Much better than the JS version. I'll pick some colours and we can 🚀

Comment (eseliger) [reply client/branded/src/search-ui/input/experimental/CodeMirrorQueryInputWrapper.tsx:242]: Props to @fkling!

Comment (eseliger) [reply client/branded/src/search-ui/input/experimental/CodeMirrorQueryInputWrapper.tsx:242]: @toolmantim to you still intend to change some colors here or would this be good to merge?

**Review by toolmantim — APPROVED** [review]

**Comment by toolmantim** [comment]
@eseliger updated the code and description. IMO this is good to go!

**Comment by toolmantim** [comment]
I'm gonna hit the big merge button.

**Pull request merged**